### PR TITLE
Add option for spatially varying mixing length parameters.

### PR DESCRIPTION
**Pull request by grmnptr**
refs #16794

### CODE REVIEW

**Comment by GiudGiud** [comment]
is the change really necessary for van Karman constants?

File: modules/navier_stokes/src/auxkernels/WallDistanceMixingLengthAux.C [hunk 1]
```diff
@@ -95,8 +95,15 @@ WallDistanceMixingLengthAux::computeValue()
...
   }
 
-  if (std::sqrt(min_dist2) / _delta <= _von_karman_const_0 / _von_karman_const)
-    return _von_karman_const * std::sqrt(min_dist2);
+  const Moose::ElemArg elem_arg = {_current_elem, false};
+  const Moose::StateArg state_arg = Moose::currentState();
+
+  const Real delta = _delta(elem_arg, state_arg);
+  const Real von_karman_const = _von_karman_const(elem_arg, state_arg);
+  const Real von_karman_const_0 = _von_karman_const_0(elem_arg, state_arg);
```
Comment (lindsayad) [inline modules/navier_stokes/src/auxkernels/WallDistanceMixingLengthAux.C:103]:
```suggestion
  const auto delta = _delta(elem_arg, state_arg);
  const auto von_karman_const = _von_karman_const(elem_arg, state_arg);
  const auto von_karman_const_0 = _von_karman_const_0(elem_arg, state_arg);
```

File: modules/navier_stokes/test/tests/finite_volume/ins/mixing_length_total_viscosity_material/tests [hunk 2]
```diff
@@ -37,5 +37,29 @@
...
+    [steady-functor-delta]
+      type = 'Exodiff'
+      input = steady.i
+      exodiff = steady-functor-delta.e
+      method = "!dbg"
+      mesh_mode = REPLICATED
+      valgrind = 'none'
+      recover = false
+      cli_args = "AuxKernels/mixing_length/delta=delta_func Outputs/file_base=steady-functor-delta"
+      detail = 'steady simulations with functions describing the boundary layer parameter'
```
Comment (lindsayad) [inline modules/navier_stokes/test/tests/finite_volume/ins/mixing_length_total_viscosity_material/tests:49]:
```suggestion
      detail = 'steady simulations with functions describing the boundary layer parameter, and'
```

File: modules/navier_stokes/test/tests/finite_volume/ins/mixing_length_total_viscosity_material/tests [hunk 3]
```diff
@@ -37,5 +37,29 @@
...
+      abs_zero = 5e-10
+    []
+    [steady-functor-delta_action]
+      type = 'Exodiff'
+      input = steady-action.i
+      exodiff = steady-functor-delta-action.e
+      method = "!dbg"
+      cli_args = "Modules/NavierStokesFV/mixing_length_delta=delta_func Outputs/file_base=steady-functor-delta-action"
+      detail = 'steady simulations with functions describing the boundary layer parameter with action syntax'
```
Comment (lindsayad) [inline modules/navier_stokes/test/tests/finite_volume/ins/mixing_length_total_viscosity_material/tests:58]:
```suggestion
      detail = 'steady simulations with functions describing the boundary layer parameter with action syntax.'
```

**Comment by grmnptr** [comment]
@GiudGiud , technically I don't think so but I'd like to give Ramiro full flexibility here.

**Review by lindsayad — APPROVED** [review]

**Comment by GiudGiud** [comment]
Test failures are unrelated timeouts

**Pull request merged**